Core/Src/main.c

```diff
@@ -87,19 +90,25 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Init(&htim1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-
+	  HAL_SPI_TransmitReceive_IT(&hspi1, &tx_buf, rx_buf, sizeof(uint8_t));
+	  *rx_buf = (uint8_t)(65535/20) + *rx_buf/20; //scales ADC digital value to be between 5-10% of 65535
+	  *rx_buf = *rx_buf & 65528; // since max value of scaled ADC value has 13 bits, we can keep most significant 10 bits
+	  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, *rx_buf); //set compare register
```

Comment:
Love the function, just maybe add a variable during your math that gets called here. When I was making this I called it "high_counts" that I passed into this. 
Having it be the save rx_buf feels weird as the name means it is meant to receive incoming data, rather than control a signal output after math conversions.